### PR TITLE
Upgrade to Slurm21.08.3

### PR DESCRIPTION
**Pull request by chenwany**
* `sbatch -G:n` is not valid in slurm21, change it to be `sbatch -G n`
* `--gpus-per-task` need to be specify with either `--gpus` or `-n/--ntasks`
* Some error message changes
    * When submit job with mutually exclusive options, error message in slurm20 is `sbatch: error: Batch job submission failed:`. Error message in slurm21 is `sbatch: error: --cpus-per-gpu is mutually exclusive with —cpus-per-task`

* job info in slurm20: `TresPerNode=gpu:{gpus_per_instance}, CpusPerTres=gpu:{cpus_per_gpu}` become `TresPerNode=gres:gpu:{gpus_per_instance}, CpusPerTres=gres:gpu:{cpus_per_gpu}` in slurm21
* Add time sleep before cluster update and after cluster update for the powering down node issue found in slurm21

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,7 +12,7 @@ X.X.X
 **CHANGES**
 - Use compute resource name rather than instance type in compute fleet Launch Template name.
 - Change SlurmQueues length and ComputeResources length schema validators to be config validators. 
-
+- Upgrade Slurm to version 21.08.3.
```
Comment (demartinofra) [inline CHANGELOG.md:15]: please leave an empty line

File: tests/integration-tests/tests/schedulers/test_slurm.py [hunk 2]
```diff
@@ -799,10 +800,10 @@ def _test_cluster_limits(slurm_commands, partition, instance_type, max_count, cp
     )
 
 
-def _submit_command_and_assert_job_rejected(slurm_commands, submit_command_args):
+def _submit_command_and_assert_job_rejected(slurm_commands, submit_command_args, reason=None):
     """Submit a limit-violating job and assert the job is failed at submission."""
     result = slurm_commands.submit_command(**submit_command_args)
-    assert_that(result.stdout).contains("sbatch: error: Batch job submission failed:")
+    assert_that(result.stdout).contains(reason if reason else "sbatch: error: Batch job submission failed:")
```
Comment (demartinofra) [inline tests/integration-tests/tests/schedulers/test_slurm.py:806]: nit: you could set reason="sbatch: error: Batch job submission failed:" as a default argument in the signature

File: tests/integration-tests/tests/schedulers/test_slurm.py [hunk 3]
```diff
@@ -996,7 +997,9 @@ def _wait_for_partition_state_changed(scheduler_commands, partition, desired_sta
 def _update_and_start_cluster(cluster, config_file):
     cluster.stop()
     _wait_for_computefleet_changed(cluster, "STOPPED")
+    time.sleep(150)
     cluster.update(str(config_file), force_update="true")
+    time.sleep(150)
```
Comment (demartinofra) [inline tests/integration-tests/tests/schedulers/test_slurm.py:1002]: please add a comment in the code explaining why this timeout is required.

File: tests/integration-tests/tests/schedulers/test_slurm.py (hunk 2 above)
Comment (chenwany) [reply tests/integration-tests/tests/schedulers/test_slurm.py]: Good idea! Changed.

File: tests/integration-tests/tests/schedulers/test_slurm.py (hunk 3 above)
Comment (chenwany) [reply tests/integration-tests/tests/schedulers/test_slurm.py]: Added

File: CHANGELOG.md (hunk 1 above)
Comment (chenwany) [reply CHANGELOG.md:15]: Added

**Review by demartinofra — APPROVED** [review]

**Pull request merged**